### ISSUE TITLE
Two suggestions on the build system

### ISSUE THREAD

**Issue by dg1727** [jcupitt/libvips#270]
I've just now compiled vips-7.42.3 (on a Debian-derived Linux) because of the statement in README.md that: "Building from git needs more packages." If you prefer I compile from the github ZIP file instead, I can try that.  

1.  I have compiled various software packages over the years, but I don't recall them needing LD_LIBRARY_PATH to contain `/usr/local/lib`.  Neither was LD_LIBRARY_PATH set to anything (for the sake of any existing package) by my current Linux distro before I compiled VIPS.  

  Is it standard that this environment variable needs to be set by the user?  If not, maybe you can address the matter in the same way as other packages.  

  It might be OK if you hard-code a search path that contains `/usr/local/lib` and is used if LD_LIBRARY_PATH isn't set or is blank.  (Because distros install to `/usr/lib`, you could have the hard-coded path contain `/usr/local/lib:/usr/lib`.)  

  Since `make install` itself is what puts files into `/usr/local/lib`, it would be only a user who moves the files who would have a search in `/usr/local/lib` fail.  There is a Filesystem Hierarchy Standard which specifies certain kinds of files (libraries, in this case) to be in the same folders on various Linux distros, so users aren't really supposed to move files away from those folder(s).  The ones who do can set LD_LIBRARY_PATH.  

2.  Ironically for this heavily multithreaded image-processing library, "make" itself is single-threaded.  Having this be multi-threaded would be terrific.

**Comment by jcupitt** [comment]
Hi @dg1727 , 

1.  There are only some circumstances or some systems where you need to set `LD_LIBRARY_PATH`. For example, if you are going to use the vips8 Python interface, gobject-introspection uses this variable to locate the shared library (I think). Some Solaris versions used to not have rpath and had to have `LD_LIBRARY_PATH` set, but perhaps that's not really relevant any more.

2. You can build with many processes, try `make -j 10`, for example.

**Comment by jcupitt** [comment]
I meant to ask, where were you reading about `LD_LIBRARY_PATH`? It sounds like some docs might need revising.

**Comment by dg1727** [comment]
When I first tried to issue, say, `vips list all` after compiling, I got an error message.  Setting `LD_LIBRARY_PATH` fixed it.

**Comment by jcupitt** [comment]
Oh, strange. What OS and version is this? You might need to run `ldconfig` after installing to `/usr/local`, it depends on how it's been set up.

**Comment by dg1727** [comment]
Xubuntu 13.04 Raring, which is unsupported.  

I simply typed `./configure`, then `make`, then (with admin privileges) `make install` as stated in `README.md`.  As noted, I've been able to do this with various other source packages and I've never had a problem with it.  

At this point, I have `LD_LIBRARY_PATH` set to `/usr/local/lib` in one of the global startup files of my shell, so things should work for me from now on.

**Comment by dg1727** [comment]
As to multithreaded `make`, you might consider adding a sentence about this to `README.md` somewhere near `By default this will install files to /usr/local.`  Shortly after that, the file talks about building on Windows & on MacOS X, but not Linux, and then goes on to "Building from git", so a Linux user who just wants to build one of the released tarfiles may not read much farther than that.  Also, the `-j` flag isn't mentioned anywhere in `README.md` at present.  

The `-j` option isn't mentioned in `INSTALL` either, so if you don't mention this option in one of the files, users would have to happen to know of it, or notice it on your website or somewhere.  The `INSTALL` file comes from the Free Software Foundation (if you change `INSTALL` and later want to import a newer version from the FSF, then you'd have to watch to preserve your changes), so I think `README.md` would be a suitable place to advise users of `make -j`.

**Comment by dg1727** [comment]
About `LD_LIBRARY_PATH`:  

The place where I got the idea to set this environment variable was http://www.vips.ecs.soton.ac.uk/index.php?title=Build_on_Ubuntu 

Based on your `ldconfig` suggestion, I tested as follows:  
1.  Comment out the `LD_LIBRARY_PATH=` statement in my shell global startup file.  
2.  Open a new shell & confirm that `LD_LIBRARY_PATH` doesn't have a value & that when I type `vips` I get the following error:  `vips: error while loading shared libraries: libvips.so.40: cannot open shared object file: No such file or directory`
3.  With admin privileges, type `ldconfig -n /usr/local/lib`.  This exited without error.  
4.  In the shell of step 2, type `vips` again.  I got the same error message.  
5.  Same result if I open a new shell and type `vips`, although I assume the new shell doesn't matter for the results of `ldconfig`.  

Setting `LD_LIBRARY_PATH` seems to work the best for now, so I'm closing this issue, but if you have any further suggestions about use of `ldconfig` I will be glad to test them.
